internal/telemetry/tracker.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package telemetry
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+	crclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+
+	"github.com/Mirantis/hmc/api/v1alpha1"
+)
+
+type Tracker struct {
+	crclient.Client
+}
+
+const interval = 10 * time.Minute
```

Comment:
@Schnitzel any preferences for the interval?

Reply:
This should be configurable, even if it's not a CRD option, but just a hidden configuration like an env var.  Also shouldn't there be an option to disable telemetry outright?  A CRD bool or something?

Reply:
@Kshatrix @eromanova 
I think 10mins is great, and I don't think we need to make it configurable interval, I can't see any other tools being able to configure the telemetry interval

@squizzi 
Yea correct we should probably do that, I created a new issue for this: https://github.com/Mirantis/hmc/issues/231